ext/pgsql/pgsql.c

```diff
@@ -2062,14 +2072,18 @@ static void php_pgsql_data_info(INTERNAL_FUNCTION_PARAMETERS, int entry_type)
 /* {{{ Returns the printed length */
 PHP_FUNCTION(pg_field_prtlen)
 {
-	php_pgsql_data_info(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_PG_DATA_LENGTH);
+	bool nullable_row = zend_string_equals_literal(EG(current_execute_data)->func->common.function_name, "pg_fieldprtlen");
+
+	php_pgsql_data_info(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_PG_DATA_LENGTH, nullable_row);
 }
 /* }}} */
 
 /* {{{ Test if a field is NULL */
 PHP_FUNCTION(pg_field_is_null)
 {
-	php_pgsql_data_info(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_PG_DATA_ISNULL);
+	bool nullable_row = zend_string_equals_literal(EG(current_execute_data)->func->common.function_name, "pg_fieldisnull");
+
+	php_pgsql_data_info(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_PG_DATA_ISNULL, nullable_row);
 }
```

Comment:
Can't we define new functions for the aliases instead of checking the name of the currently called function.

Anyway that can be a follow-up optimization.

Reply:
Hm, yes, I guess we could easily split the aliases into two functions in these cases, since the implementations are a one-liner.